lib/pdf/external_font.ex

```diff
@@ -118,38 +139,22 @@ defmodule Pdf.ExternalFont do
     iex> ExternalFont.width(font, "A")
     123
   """
-  def width(font, <<char_code::integer>> = str) when is_binary(str) do
-    width(font, char_code)
-  end
-
   def width(font, char_code) do
-    Pdf.Encoding.WinAnsi.characters()
```

Comment:
What about building an internal map to handle WinAnsi characters only and putting the rest on ETS?
Or, throwing away the rest for now because we’re not able to handle UTF-8 yet?

Reply:
Aren't we doing that now when I fill up the `ets` table?
My issue with the previous lookup method was that the test suite for one of my projects took > 20s, when using an external font, vs 8s when using Helvetica. With these changes it goes down to ~8.9s. 
I don't know what is the faster solution, `ets` or a lookup in a `Map`.

Reply:
The full external font is being loaded into the map which means it contains every glyph and kerning pair contained in the external font.
My suggestion is that we only load the glyphs and kern pairs that belong to the WinAnsi encoding.
Can you send me the font test with the external font so I can see the performance differences.

Reply:
I set up a test [repo](https://github.com/Hermanverschooten/andrew_pdf).
If I run this with the current version (0.5), it takes about 5.4s, with the ets version 0.6a.

Reply:
Even faster ;-)
Try this: https://github.com/andrewtimberlake/elixir-pdf/commit/590420901c099533b3863e682038426f90eca7d7

```
time mix test
...

Finished in 1.7 seconds
1 doctest, 2 tests, 0 failures

Randomized with seed 452380

real    0m2.665s
user    0m2.900s
sys     0m0.680s
```

Reply:
ets version with your patch:

time mix test
...

Finished in 0.5 seconds
1 doctest, 2 tests, 0 failures

Randomized with seed 784146
mix test  1,67s user 0,44s system 167% cpu 1,255 total

Reply:
I’m still not convinced that we need to bring ets in just yet.
I’ve done some further optimisation by moving the iteration over the character map to pre-build a character map of widths rather than iterating over the characters for each call to ExternalFont.width/1
https://github.com/andrewtimberlake/elixir-pdf/commit/3026341f24a2745ea1d91f69acd3782cbbc09e11

Reply:
We do seem to have lost the `€` now. That gives a
```
 ** (ArithmeticError) bad argument in arithmetic expression
```

Reply:
It was returning `nil` when no width could be found. I have fixed that error in 4ddf72e but do you have a width for € in your font that it’s not finding?

Reply:
Strange, now no error and the `€` is in the PDF.

Reply:
Maybe close this PR and continue with your branch then. I don't mind.

Reply:
BTW the test suite of the whole project now differs by about 1 second, 8.1s vs 9.5s now, vs 20.5s before.